types/module/module.go

```diff
@@ -298,7 +300,18 @@ func (m *Manager) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, genesisData
 		}
 		ctx.Logger().Debug("running initialization for module", "module", moduleName)
 
-		moduleValUpdates := m.Modules[moduleName].InitGenesis(ctx, cdc, genesisData[moduleName])
+		// Get the type of the default genesis so we can create a zeroed struct
+		// to which we then unmarshal the genesis' file JSON into
+		genType := reflect.TypeOf(m.Modules[moduleName].DefaultGenesis())
+		if genType == nil {
+			continue
```

Comment:
I would panic here instead TBH. It's feedback to the app developers that their DefaultGenesis is incorrect.

Reply:
For example `x/auth/vesting` and `x/upgrade` returned this:

```
func (AppModuleBasic) DefaultGenesis(_ codec.JSONCodec) json.RawMessage {
	return []byte("{}")
}
```